app/models/rook.rb

```diff
@@ -1,2 +1,11 @@
 class Rook < Piece
+  def valid_move?(new_x, new_y)
+    unless is_obstructed?(new_x, new_y)
+      return true if (y_position == new_y) && (x_position != new_x) ||
+        (x_position == new_x) && (y_position !=new_y)
+    end 
+  end
 end
+    
+
```

Comment:
Damn it, I really didn't mean to erase your previous code, I was just trying to add what I came up with for the rook move.  Sorry.  Trying to find out how to get it back.